server/bin/pbench-index.py

```diff
@@ -40,6 +48,236 @@
+
+    # We set the chunk size to 50 MB which is half the maximum payload size
+    # that Elasticsearch typically handles.
+    _CHUNK_SIZE = 50 * 1024 * 1024
+
+    def __init__(
+        self,
+        config,
+        name,
+        es=None,
+        pid=None,
+        group_id=None,
+        user_id=None,
+        hostname=None,
+        version=None,
+        templates=None,
+    ):
+        self.config = config
+        self.name = name
+        self.logger = pbench.get_pbench_logger(name, config)
+
+        # We always create a base "tracking" document composed of parameters
+        # from the caller, and other environmental data. This document is used
+        # as the foundation for the first document posted to the target
+        # Elasticsearch instance with the `post_status()` method.  All
+        # subsequent calls to the `post_status()` method will use that first
+        # document ID as their parent document ID.  This allows us to have
+        # multiple status updates associated with the initial Report() caller.
+        if config._unittests:
+            _hostname = "example.com"
+            _pid = 42
+            _group_id = 43
+            _user_id = 44
+        else:
+            _hostname = hostname if hostname else socket.gethostname()
+            _pid = pid if pid else os.getpid()
+            _group_id = group_id if group_id else os.getgid()
+            _user_id = user_id if user_id else os.getuid()
+        self.generated_by = dict(
+            [
+                ("commit_id", self.config.COMMIT_ID),
+                ("group_id", _group_id),
+                ("hostname", _hostname),
+                ("pid", _pid),
+                ("user_id", _user_id),
+                ("version", version if version else ""),
+            ]
+        )
+        # The "tracking_id" is the final MD5 hash of the first document
+        # indexed via the `post_status()` method.
+        self.tracking_id = None
+        try:
+            self.idx_prefix = config.get("Indexing", "index_prefix")
+        except (NoOptionError, NoSectionError):
+            # No index prefix so reporting will be performed via logging.
+            self.idx_prefix = None
+            self.es = None
+        else:
+            if es is None:
+                try:
+                    self.es = get_es(config, self.logger)
+                except Exception:
+                    self.logger.exception(
+                        "Unexpected failure fetching" " Elasticsearch configuration"
+                    )
+                    # If we don't have an Elasticsearch configuration just use
+                    # None to indicate logging should be used instead.
+                    self.es = None
+            else:
+                self.es = es
+        if templates is not None:
+            self.templates = templates
+        else:
+            self.templates = PbenchTemplates(
+                self.config.BINDIR, self.idx_prefix, self.logger
+            )
```

Comment:
This code is a lift-n-shift.  And it will be deleted entirely in PR #3293.

Though we should probably remove the reporting mechanism from the production code as well.